design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,306 @@
+- As a KubeVirt cluster operator, I would like to stop using a feature
+  that got removed.
+
+### Repos
+This is a cross repo project policy under the
+[kubevirt](https://github.com/kubevirt) organization.
+
+## Proposal Design
+The proposal on how to define a feature lifecycle is influenced by
+processes and policies from the Kubernetes project.
+These sources are scattered around, each focusing on different
+aspects of a feature:
+- [Feature Gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/)
+- [Graduation](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/)
+- [Changing the API](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md)
+- [Deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+The proposal takes the top-down approach, starting with the high level
+flow that a common feature will traverse through.
+Continuing with actions that need to be taken and timeline suggestions.
+
+Both feature graduation and discontinuation flows are covered.
+Including the implications on users.
+
+Depending on individual topics, follow-up proposals may extend the basic
+points raised in this proposal.
+
+### Feature Stages
+A feature is expected to pass through the following stages:
+1. Enhancement proposal.
+2. Implementation.
+3. Release as Alpha (experimental).
+4. Release as Beta (pre-release for evaluation).
+5. Release as General Availability (graduation).
+6. Removal.
+
+Starting from the Alpha release, it can be removed with restrictions that
+depend on the release stage (Alpha, Beta, GA).
+
+[Removal](#removal) of features is widely discussed later
+in this proposal.
+
+#### Enhancement proposal
+As the first step for introducing a new feature, a formal proposal is
+expected to be shared for public review via mailinglist and
+a [design proposal](https://github.com/kubevirt/community/tree/main/design-proposals).
+
+This is the first opportunity to evaluate a new feature.
+The proposal needs to include motivation, goals, implementation details
+and phases. Review the [proposal template](https://github.com/kubevirt/community/blob/main/design-proposals/proposal-template.md)
+for more information.
+
+#### Implementation
+The development work on the feature is expected to include coding,
+testing, integration and documentation.
+
+While most of the effort is expected to occur towards the Alpha stage,
+work may start as a PoC during the design and as adjustments between
+the release stages.
+
+#### Releases
+- **Alpha**:
+  An initial release of the feature for experimental purposes.
+  Recommended for non-production usages, evaluation or testing.
+
+  The API is considered unstable and may change significantly.
+
+  The period in which a feature can remain in Alpha is limited,
+  assuring features are not piling up without control.
+
+  The feature functionality is controlled using a Feature-Gate (FG) during
```

Comment:
Can we name it "feature presence"?

As the FG is actually not impacting the functionality, it's only controlling the presence of the feature.